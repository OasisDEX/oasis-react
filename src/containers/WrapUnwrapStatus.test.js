/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { fromJS } from 'immutable';

import {
  WrapUnwrapStatusWrapper,
  mapStateToProps,
  mapDispatchToProps, WRAP_STATUS_VIEW_TYPE_WRAP,
} from './WrapUnwrapStatus';
import { shallow } from 'enzyme';

describe('(Container) WrapUnwrapStatus', () => {
  const state = fromJS(global.storeMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps,
    type: WRAP_STATUS_VIEW_TYPE_WRAP
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <WrapUnwrapStatusWrapper {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
