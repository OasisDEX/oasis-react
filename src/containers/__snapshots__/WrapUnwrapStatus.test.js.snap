// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) WrapUnwrapStatus should render 1`] = `""`;

exports[`(Container) WrapUnwrapStatus will receive right actions 1`] = `
Object {
  "actions": Object {},
}
`;

exports[`(Container) WrapUnwrapStatus will receive right props 1`] = `
Object {
  "activeTokenUnwrapStatus": null,
  "activeTokenWrapStatus": null,
}
`;
